[kokerinks] iP

kokerinks opened the pull request:
# DukePro

> This is a quote - By me

DukePro frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/admin/ip-w4.html)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 👍 

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

kokerinks committed:
Level 0. Rename, Greet, Exit

kokerinks committed:
Level 1. Echo

kokerinks committed:
Level 2. Add, List

kokerinks committed:
Level 3. Mark as Done

kokerinks committed:
Level 4. ToDos, Events, Deadlines

kokerinks committed:
A-TextUiTesting: Automated Text UI Testing

kokerinks committed:
Level 5. Handle Errors

kokerinks committed:
Level 5 & Level 6

kokerinks committed:
Level 7. Save

kokerinks committed:
Level 8. Dates and Times

kokerinks committed:
Add classes Ui, Command, Parser, Storage, Tasklist

kokerinks committed:
Refactor Duke to use new classes

kokerinks committed:
Move classes into packages duke and duke.task

kokerinks committed:
Merge remote-tracking branch 'origin/add-gradle-support'

kokerinks committed:
Change code to be Java 11 compatible

kokerinks committed:
Add new valid dates into datetime formats

kokerinks committed:
Add JUnits test classes for Todo and Deadline

kokerinks committed:
Alter settings for build.gradle

kokerinks committed:
Add JavaDocs for all classes

kokerinks committed:
Add find feature for Level-9

kokerinks committed:
Merge branch 'branch-Level-9'

kokerinks committed:
Change code to follow Java coding standard

kokerinks committed:
Merge branch 'branch-A-CodingStandard'

kokerinks committed:
Add introductory javafx functionality

kokerinks committed:
Add rudimentary GUI functionality using javaFX

kokerinks committed:
Fix bugs with GUI functionality

kokerinks committed:
Fix saving of file after every command

howenyap commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,40 @@
+     *
+     * @param newType Type of the Command
+     */
+    public Command(Duke.CommandType newType) {
```
Maybe you can rename 'newType' as 'type' because this.type already differentiates between class fields and method parameters?

howenyap commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,40 @@
+     * @param newType Type of the Command
+     * @param newArgs Arguments required for running of command
+     */
+    public Command(Duke.CommandType newType, String[] newArgs) {
```
Same goes for 'newArgs' here

howenyap commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,70 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
```
Great job following import standards!

howenyap commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,348 @@
+    @Override
+    public void start(Stage stage) {
+
+        //Step 1. Setting up required components
```
You can add a whitespace after '//' for better visibility!

howenyap commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,136 @@
+     */
+    public Command parse(String input) throws DukeException {
+        String[] parts = input.split(" ");
+        switch (parts[0]) {
```
There shouldn't be indentation for `case` clauses here

howenyap reviewed (commented):
Great coding habits overall!

kokerinks committed:
Add build.gradle settings for fat JAR generation

kokerinks committed:
Add relevant assertions for some classes

kokerinks committed:
Edit the code to improve code quality

Code was editted based in SE-EDU initiative
to improve code quality.

Changes include:
- Split parse method in Parser into smaller methods
- Add spaces for comments in Duke.java for readability
- Remove unused statements in MainWindow.java

kokerinks committed:
Merge pull request #1 from kokerinks/branch-A-Assertions

Add relevant assertions for some classes

kokerinks committed:
Merge branch 'master' into branch-A-CodeQuality

kokerinks committed:
Merge pull request #2 from kokerinks/branch-A-CodeQuality

Improved code quality based on Java SE documentation

kokerinks committed:
Add gradle.yml for GitHub Actions

kokerinks committed:
Add sorting functionality to list command

minreiseah commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,349 @@
+        scrollPane.setVvalue(1.0);
+        scrollPane.setFitToWidth(true);
+
+        // You will need to import `javafx.scene.layout.Region` for this.
```
Perhaps remove these comments since the relevant components are already imported above.

minreiseah commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,349 @@
+        return textToAdd;
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Should be proper javadoc here, not github copilot 🧐

minreiseah commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,349 @@
+    public String getResponse(String input) {
+        try {
+            Command cmd = parser.parse(input);
+            assert cmd != null : "Command should not be null";
```
It's considered best practice to use assert for testing/debugging. Considering how you already have a default clause for invalid command inputs, perhaps use that to handle empty commands?

minreiseah reviewed (approved):
Nice work! I've been inspired by your clean coding standards.

Hibeom0929 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+    /**
+     * List of tasks
+     */
+    protected ArrayList<Task> tasks;
```
Good naming for multi-valued variables!

Hibeom0929 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,78 @@
+    /**
+     * Whether the Task is done
+     */
+    protected boolean isDone;
```
Good naming for the boolean variable.

Hibeom0929 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,158 @@
+            case "delete":
+                return parseDelete(input);
+            case "find":
+                return parseFind(input);
```
The indentation of the case should be the same as the switch

Hibeom0929 reviewed (commented):
Overall LGTM! All of your naming is appropriate and used standard words. There is no confusion with the meaning of variables because of the naming.

kokerinks committed:
Rename all Duke names to Kokbot

kokerinks committed:
Add Javadocs for MainWindow

kokerinks committed:
Remove dead code from JavaFX tutorial

kokerinks committed:
Add Javadocs for DialogBox and Parser

kokerinks committed:
Add JUnit testcases for Parser, Deadline, Todo

kokerinks committed:
Add Ui.png to docs

kokerinks committed:
Refactor Kokbot package to kokbot

kokerinks committed:
Fix help message for LIST command

kokerinks committed:
Add README.md to /docs

kokerinks committed:
Fix links on README.md
